### PR TITLE
Move Filter* to sub folders

### PR DESCRIPTION
**Pull request by gyfchong**
Grouping like-components together

### CODE REVIEW

File: packages/components/src/Filter/FilterButton/FilterButtonRemovable/FilterButtonRemovable.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Tooltip } from "@kaizen/draft-tooltip"
 import { ClearIcon } from "~components/Icon"
 import { DataAttributes } from "~types/DataAttributes"
 import { isRefObject } from "~utils/isRefObject"
-import { ButtonGroup, ButtonGroupProps } from "../../ButtonGroup"
+import { ButtonGroup, ButtonGroupProps } from "../../../ButtonGroup"
```
Comment (HeartSquared) [inline packages/components/src/Filter/FilterButton/FilterButtonRemovable/FilterButtonRemovable.tsx:6]:
```suggestion
import { ButtonGroup, ButtonGroupProps } from "~components/ButtonGroup"
```

File: packages/components/src/Filter/FilterButton/subcomponents/FilterButtonBase/FilterButtonBase.module.scss [hunk 2]
```diff
@@ -2,7 +2,7 @@
 @import "~@kaizen/design-tokens/sass/color";
 @import "~@kaizen/design-tokens/sass/typography";
 @import "~@kaizen/design-tokens/sass/spacing";
-@import "../../../../styles/button-reset";
+@import "../../../../../styles/button-reset";
```
Comment (HeartSquared) [inline packages/components/src/Filter/FilterButton/subcomponents/FilterButtonBase/FilterButtonBase.module.scss:5]: OOS: Might be nice if we made some aliases for these lol

File: packages/components/src/Filter/FilterDateRangePicker/subcomponents/FilterDateRangePickerField/utils/validateEndDateBeforeStartDate.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React from "react"
-import { DateValidationResponse } from "~components/FilterDatePicker"
-import { getNodeText } from "../../../../utils/getNodeText"
+import { DateValidationResponse } from "~components/Filter/FilterDatePicker"
+import { getNodeText } from "../../../../../utils/getNodeText"
```
Comment (HeartSquared) [inline packages/components/src/Filter/FilterDateRangePicker/subcomponents/FilterDateRangePickerField/utils/validateEndDateBeforeStartDate.ts:3]:
```suggestion
import { getNodeText } from "~utils/getNodeText"
```

File: packages/components/src/Filter/FilterSelect/FilterSelect.tsx [hunk 4]
```diff
@@ -19,6 +18,7 @@ import {
 import { getDisabledKeysFromItems } from "~components/__future__/Select/utils/getDisabledKeysFromItems"
 import { transformSelectItemToCollectionElement } from "~components/__future__/Select/utils/transformSelectItemToCollectionElement"
 import { OverrideClassName } from "~types/OverrideClassName"
+import { Filter, FilterContents } from "../Filter"
```
Comment (HeartSquared) [inline packages/components/src/Filter/FilterSelect/FilterSelect.tsx:21]: Oh? Don't want the alias for this one?

File: packages/components/src/Filter/FilterButton/subcomponents/FilterButtonBase/FilterButtonBase.module.scss (hunk 2 above)
Comment (gyfchong) [reply packages/components/src/Filter/FilterButton/subcomponents/FilterButtonBase/FilterButtonBase.module.scss:5]: I think the only way we can alias these is if we made it its own package, cause I don't think SCSS has an alias function 🤔

File: packages/components/src/Filter/FilterBar/context/utils/checkShouldUpdateValues.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { FiltersValues } from "~components/FilterBar/types"
+import { FiltersValues } from "~components/Filter/FilterBar/types"
```
[the remark this answers is not in the document]
Comment (gyfchong) [reply packages/components/src/Filter/FilterBar/context/utils/checkShouldUpdateValues.ts:1]: Lol this is pretty confusing:
```
import { FiltersValues } from "../../types"
import { FilterBarState } from "../types"
```

File: packages/components/src/Filter/FilterSelect/FilterSelect.tsx (hunk 4 above)
Comment (gyfchong) [reply packages/components/src/Filter/FilterSelect/FilterSelect.tsx:21]: Weird, its an alias for me lol

File: packages/components/src/Filter/FilterButton/subcomponents/FilterButtonBase/FilterButtonBase.module.scss (hunk 2 above)
Comment (HeartSquared) [reply packages/components/src/Filter/FilterButton/subcomponents/FilterButtonBase/FilterButtonBase.module.scss:5]: Pretty sure it does, but where the config lives is the question 😬 For another day

File: packages/components/src/Filter/FilterDatePicker/_docs/FilterDatePicker.stories.tsx [hunk 6]
```diff
@@ -3,12 +3,12 @@ import { Meta, StoryObj } from "@storybook/react"
 import Highlight from "react-highlight"
 import { Paragraph } from "@kaizen/typography"
 import { defaultMonthControls } from "~components/Calendar/_docs/controls/defaultMonthControls"
-import { renderTriggerControls } from "~components/Filter/_docs/controls/renderTriggerControls"
+import { renderTriggerControls } from "~components/Filter/Filter/_docs/controls/renderTriggerControls"
```
Comment (HeartSquared) [inline packages/components/src/Filter/FilterDatePicker/_docs/FilterDatePicker.stories.tsx:6]: Oh actually these shared controls can now move to the top level :) It was actually just shared stuff for the `Filter*` components, and not for the `Filter` component itself.

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**